driver-core/src/main/com/mongodb/connection/SocketSettings.java

```diff
@@ -93,25 +95,27 @@ public Builder applySettings(final SocketSettings socketSettings) {
         /**
          * Sets the socket connect timeout.
          *
-         * @param connectTimeout the connect timeout
+         * @param connectTimeout the connect timeout.
+         * The timeout converted to milliseconds must not be greater than {@link Integer#MAX_VALUE}.
          * @param timeUnit the time unit
          * @return this
          */
-        public Builder connectTimeout(final int connectTimeout, final TimeUnit timeUnit) {
-            this.connectTimeoutMS = MILLISECONDS.convert(connectTimeout, timeUnit);
+        public Builder connectTimeout(final long connectTimeout, final TimeUnit timeUnit) {
+            this.connectTimeoutMS = toIntExact(MILLISECONDS.convert(connectTimeout, timeUnit));
```

Comment:
Would this be a new thing to have the driver throw `ArithmeticException` on an invalid value from the application?  Should the exception instead be wrapped in `IllegalArgumentException`, making it similar to `ConnectionString`?

Reply:
Thank you, done.